### ISSUE TITLE
file_error_alert "not supported"

### ISSUE THREAD

**Issue by btowntkd** [linuxserver/docker-deluge#139]
I'm not sure exactly when this happened, but sometime a couple days ago, my deluge docker updated and my torrents all started failing.  I'm running Docker for Windows Desktop using WSL2. As soon as I add a torrent, it moved to "ERROR" state as soon as it receives any data whatsoever.

When I enable DEBUG logging, I see the following example error:
```
08:00:01 [DEBUG   ][deluge.core.alertmanager      :130 ] file_error_alert: slackware-14.2-iso file_open (/downloads/Incomplete/torrent/slackware-14.2-iso/slackware-14.2-install-dvd.iso) error: Not supported
08:00:01 [DEBUG   ][deluge.core.alertmanager      :135 ] Handling alert: file_error_alert
08:00:01 [DEBUG   ][deluge.core.alertmanager      :130 ] torrent_error_alert: slackware-14.2-iso ERROR: (95 Not supported) /downloads/Incomplete/torrent/slackware-14.2-iso/slackware-14.2-install-dvd.iso
08:00:01 [DEBUG   ][deluge.core.alertmanager      :130 ] torrent_paused_alert: slackware-14.2-iso paused
08:00:01 [DEBUG   ][deluge.core.alertmanager      :135 ] Handling alert: torrent_paused_alert
```

Any idea what might have changed, recently?

**Comment by delamoe** [comment]
I'm having the same issue, all torrents fail after a few seconds and the GUI shows them in an error state. Reverting to an older tag fixed for me. `deluge:amd64-2.0.5-0202202181752ubuntu20.04.1-ls140`
This is what the errors looked like:
```
deluge  | 16:04:42 [DEBUG][deluge.core.alertmanager:135 ] Handling alert: file_error_alert
deluge  | 16:04:42 [DEBUG][deluge.core.alertmanager:130 ] torrent_error_alert: <filename> ERROR: (19 No such device) /media/downloads/deluge/incomplete/<filename>
deluge  | 16:04:42 [DEBUG][deluge.core.alertmanager:130 ] file_error_alert: <filename> file_open (/media/downloads/deluge/incomplete/<filename>) error: No such device
```

**Comment by thelamer** [comment]
If you are using mergefs: 
https://github.com/linuxserver/docker-deluge/issues/140#issuecomment-1059799488

**Comment by btowntkd** [comment]
> If you are using mergefs: [#140 (comment)](https://github.com/linuxserver/docker-deluge/issues/140#issuecomment-1059799488)

The mount in question is an NTFS drive connected via Docker for WIndows Desktop + WSL2.  Following delamoe's suggestion of reverting to a previous tag also worked as a temporary fix for me.

**Comment by thelamer** [comment]
Can anyone having this issue do me a favor and try passing `--security-opt seccomp=unconfined` when running the latest container ?

**Comment by btowntkd** [comment]
It does not appear to change the problem.  For sanity's sake, this is my docker-compose with the suggested tweak:

```
deluge:
    image: linuxserver/deluge
    container_name: deluge
    network_mode: service:vpn
    depends_on:
      - vpn
     security_opt:
       - seccomp:unconfined
    volumes:
      - ${APPDATA_PATH}/deluge:/config
      - ${MEDIA_PATH}:/media
    restart: unless-stopped
    environment:
      - TZ=${TZ}
      - PUID=${PUID}
      - PGID=${PGID}
```

**Comment by thelamer** [comment]
Bummer, it was worth a shot. It seems like running a more head distro including libtorrent etc breaks non standard volume mounts. The fact that it has problems with specific mergefs setups made me think specific sys calls were not working due to a libseccomp mismatch. 
I don't have a "this will fix it for you" update, but I am fairly confident we will not be rolling back. Will leave this open as a catch all if anyone has an epiphany out there.

**Comment by ShauneBoy** [comment]
Not sure if this is relevant but I also use qBitTorrent (as a docker image) but any build after 4.3.9 gives me the same error - so I thought would switch over to Deluge - low and behold, I am getting the same error... 4.3.9 of qBitTorrent docker image seems to be built on ubuntu - will try installing an older version on deluge [as previous comments say it is fixed] so is it related to something in the os build or something? [sorry, I'm somewhat of a *nix newbie - understand it enough but am no expert in it]

**Comment by thelamer** [comment]
@ShauneBoy hop on discord for support, this issue is open to start to catalogue the storage configurations that are causing this error.
So far we have:
Mergefs with directio enabled 
And WSL NTFS volume mount.

Please post yours.

**Comment by medsouz** [comment]
I'm currently experiencing this as well with BTRFS over NFS.

**Comment by delamoe** [comment]
My storage config is BTRFS via a unionfs mountpoint

**Comment by delamoe** [comment]
The latest tag works as expected when I mount the BTRFS mountpoint directly. I suspect that unionfs is part of the error. I don't need to utilize the merged filesystem for Deluge so this is a fix for me.

**Comment by ShauneBoy** [comment]
> @ShauneBoy hop on discord for support, this issue is open to start to catalogue the storage configurations that are causing this error. So far we have: Mergefs with directio enabled And WSL NTFS volume mount.
> 
> Please post yours.

@thelamer ok will do... as to my config, it's ext4 - although it is on a USB device (/dev/mmcblk0) mounted using 'by-uuid' in /etc/fstab [ubuntu 20.04.4 LTS] (and not sure if this helps but get the same issue using LinuxServer qBitTorrent docker image on anything higher than the last 4.3.9 build)

**Comment by thelamer** [comment]
So I still have not been able to replicate this on my local rig, but I do run an edge disto and btrfs. I need to spin up a VM to really test stuff. 
I suspect that these issues might steam from libtorrents `fallocate` call when downloading a torrent. 

Can anyone in this thread test setting "Pre-allocate disk space" in their settings and see if this helps? (please restart the container after making the change just to really make sure it took hold)

**Comment by ShauneBoy** [comment]
> Can anyone in this thread test setting "Pre-allocate disk space" in their settings and see if this helps? (please restart the container after making the change just to really make sure it took hold)

I do have this set (in my current setup) - the file gets created correctly and to the correct size (at the correct location /downloads) - it is as soon as the torrent starts to receive data, it drops to an 'Error' status...

**Comment by delamoe** [comment]
> > Can anyone in this thread test setting "Pre-allocate disk space" in their settings and see if this helps? (please restart the container after making the change just to really make sure it took hold)
> 
> I do have this set (in my current setup) - the file gets created correctly and to the correct size (at the correct location /downloads) - it is as soon as the torrent starts to receive data, it drops to an 'Error' status...

I can confirm the same error on my setup with the unionfs mount

**Comment by thelamer** [comment]
Ok so not falloc. Just breaking down what is occurring we have two documented errors 19 and 95. 
95 would indicate that setting a permission is not supported on the file it is creating. 
19 generally indicates a device access error more specifically network device. For no such device errors is there anything non standard about the network stack?

**Comment by ShauneBoy** [comment]
> Reporting the same issue, the last update of the linuxserver/deluge image reports errors and the webui and api goes down regularly. Reverting to tag `deluge:amd64-2.0.5-0202202181752ubuntu20.04.1-ls140` fixes these issues.

Ok, just tried this version [although had to use tag deluge:2.0.5-0202202181752ubuntu20.04.1-ls140] and confirm it works [no errors - downloads are fine]... Got to wonder if some issue with Alpine? (don't know enough the differences here) as I also use linuxserver/qBitTorrent and had to drop back to 4.3.9 (which was the last ubuntu release) - If I pick *any* release that is Alpine, I get exactly the same issue on that as well [ie it creates and sizes the file ok but any subsequent write to the file fails]

**Comment by EzekialSA** [comment]
can confirm same issues/solutions from latest CoreOS branch, reverting back to the ubuntu branch mentioned resolved my issues. Was annoying since there were no logs on the alpine branch that alluded to any issues, just errors. I also didn't see any SELinux alerts in the audit.log for this.

**Comment by madhavtummala** [comment]
I am facing this issue on my Raspi4B running ubuntu server arm64. The download location is set to a mounted NTFS drive. Reverting to "2.0.5-0202202181752ubuntu20.04.1-ls140" solves to problem

**Comment by courtarro** [comment]
Checking in with the same issue. NFS connected to a ZFS array.

I downgraded to `2.0.5-0202202181752ubuntu20.04.1-ls140`, but afterward I had to move my `session.state` file and Force Recheck all my existing torrents, just in case anyone else runs into the same issues after the downgrade.

**Comment by thelamer** [comment]
From what I have been told if you turn off pre allocation and re add a torrent it should work. 
Process to test would be to turn off pre allocation, clear out current torrents and then add a fresh one after restarting the container.

**Comment by stuartraetaylor** [comment]
I had the same problem with a mergerfs (2.32.6) filesystem on Ubuntu server.

The error reported by Deluge:
```
18:09:48 [DEBUG   ][deluge.core.alertmanager      :130 ] torrent_error_alert: 2020-02-13-raspbian-buster.zip ERROR: (19 No such device) /downloads/2020-02-13-raspbian-buster.zip
```

Setting the docker volume to point directly to an ext4 filesystem mitigated the problem, as did reverting the docker image to 2.0.5-0202202181752ubuntu20.04.1-ls140.

Finally removing the `direct_io` option from the mergerfs filesystem resolved the issue using the latest docker image.

The problem appears to be caused by a known issue with FUSE and mmap:
* [mergerfs: rtorrent fails with ENODEV (No such device)](https://github.com/trapexit/mergerfs#rtorrent-fails-with-enodev-no-such-device)
* [mergerfs: CACHING](https://github.com/trapexit/mergerfs#caching)

**Comment by kamakamadaun** [comment]
I am having the same problem. I am running deluge in docker on windows/WSL2 and saving to a mounted network drive (smb).
I would like not to have to save locally first.
It seems that there are other work-arounds above (mount directly, direct_io) but I don't understand what that would mean specifically. Would you be so kind and write exactly how I would do what you suggest?

**Comment by ITestInProd** [comment]
Seeing this as well with the current 2.1.1 image based upon Alpine. The referenced image above, based upon Ubuntu, does not experience a move error.

I suspect this may be an upstream libtorrent v2.x error. A similar "moving upon completion error" was reported there as well- #https://github.com/arvidn/libtorrent/issues/6985.

**Comment by download13** [comment]
> From what I have been told if you turn off pre allocation and re add a torrent it should work. Process to test would be to turn off pre allocation, clear out current torrents and then add a fresh one after restarting the container.

Tried this and it worked for me. I'm guessing that means it is a pre-allocation issue. I'm using an Ubuntu 19.10 VM with a CIFS mount for the downloads directory.

That seems like it would indicate that some change between 2.0.5 and 2.1.1 broke pre-allocation across CIFS boundaries.

**Comment by instantdreams** [comment]
Updated compose to use `lscr.io/linuxserver/deluge:latest` which pulled v 2.1.1.

Tested and is working.

**Comment by mueslo** [comment]
I also ran into this in a fresh setup using Docker with NFS volume mount (ZFS remote) on the latest v2.1.1 image.

Disabling pre-allocation fixed it for me also. However, that is a workaround and not a solution. The referenced libtorrent issue was fixed in libtorrent 2.0.8, so it must be a different, possibly related issue.

Also the fact that such an obvious (libtorrent) failure with "Error" in the front-end of deluge does not produce any log entry or further information beyond a DEBUG log entry should be considered a bug.
